inc/namespace.php

```diff
@@ -357,3 +364,103 @@ function output_integrity_for_style( string $html, string $handle ) : string {
 function send_xss_header() {
 	header( 'X-XSS-Protection: 1; mode=block' );
 }
+
+/**
+ * Filter an individual policy value.
+ *
+ * @param string $name Directive name.
+ * @param string|array $value Directive value.
+ * @return string[] List of directive values.
+ */
+function filter_policy_value( string $name, $value ) : array {
+	$value = (array) $value;
+
+	$needs_quotes = [
+		'self',
+		'unsafe-inline',
+		'unsafe-eval',
+		'none',
+		'strict-dynamic',
+	];
+
+	// Normalize directive values.
+	foreach ( $value as &$item ) {
+		if ( in_array( $item, $needs_quotes, true ) || strpos( $item, 'nonce-' ) === 0 ) {
+			// Add missing quotes if the value was erroneously added
+			// without them.
+			$item = sprintf( "'%s'", $item );
+		}
+	}
+
+	/**
+	 * Filter value for a given policy directive.
+	 *
+	 * `$name` is the directive name.
+	 *
+	 * @param array $value List of directive values.
+	 */
+	$value = apply_filters( "altis.security.browser.filter_policy_value.$name", $value );
+
+	/**
+	 * Filter value for a given policy directive.
+	 *
+	 * @param array $value List of directive values.
+	 * @param string $name Directive name.
+	 */
+	return apply_filters( 'altis.security.browser.filter_policy_value', $value, $name );
+}
+
+/**
+ * Send the Content-Security-Policy header.
+ *
+ * The header is only sent if policies have been specified. See
+ * get_content_security_policies() for setting the policies.
+ */
+function send_csp_header() {
+	// Gather and filter the policy parts.
+	$policies = get_content_security_policies();
+	$policy_parts = [];
+	foreach ( $policies as $key => $value ) {
+		$value = filter_policy_value( $key, $value );
+		if ( empty( $value ) ) {
+			continue;
+		}
+		$policy_parts[] = sprintf( '%s %s', $key, implode( ' ', $value ) );
```

Comment:
Is there any sanitisation of `$value` we can or should be doing?

Reply:
Not really; it's not user input, and the values can be practically anything.